pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1127,7 +1127,17 @@ func (ctrl *PersistentVolumeController) recycleVolumeOperation(volume *v1.Persis
 		glog.V(3).Infof("can't recycle volume %q: %v", volume.Name, err)
 		return
 	}
-	if used {
+
+	// Verify the claim is in cache: if so, then it is a different PVC with the same name
+	// since the volume is know to be released at this moment
+	claimName := claimrefToClaimKey(volume.Spec.ClaimRef)
```

Comment:
Please add some comment why it's ok to recycle the volume when there is a PVC in cache.
"There is a new PVC with the same name as the deleted one. The new PVC must use a different PV, because we checked that the PV is unused in isVolumeReleased. So the old PV is safe to be recycled."

Reply:
Yessir.